src/Gaufrette/Adapter/AzureBlobStorage.php

```diff
@@ -308,6 +310,26 @@ public function mtime($key)
 
     /**
      * {@inheritdoc}
+     */
+    public function size($key)
+    {
+        $this->init();
+        list($containerName, $key) = $this->tokenizeKey($key);
+
+        try {
+            $properties = $this->blobProxy->getBlobProperties($containerName, $key);
+
+            return $properties->getProperties()->getContentLength();
+        } catch (ServiceException $e) {
+            $this->failIfContainerNotFound($e, sprintf('read content length for key "%s"', $key), $containerName);
+
+            return false;
```

Comment:
Be careful, neither the [`SizeCalculator`](https://github.com/KnpLabs/Gaufrette/blob/master/src/Gaufrette/Adapter/SizeCalculator.php#L17) nor the [`FilesystemInterface`](https://github.com/KnpLabs/Gaufrette/blob/master/src/Gaufrette/FilesystemInterface.php#L138) have a `bool` return value. Maybe you throw a `\RuntimeException` here instead of returning false ?

Reply:
Actually, only `FTP` and `AwsS3` implements this interface and both differs: the former throws an exception whereas the latter returns false... So I think we can consider this acceptable and it'll be replaced by a proper exception when backported to 1.0 branch.

Reply:
:+1: for proper impl in 1.0